src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -276,5 +276,11 @@ public override int GetHashCode(string obj)
         // Equals/GetHashCode methods for the comparer itself. 
         public override bool Equals(object obj) => obj is OrdinalIgnoreCaseComparer;
         public override int GetHashCode() => nameof(OrdinalIgnoreCaseComparer).GetHashCode();
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            info.SetType(typeof(OrdinalComparer));
+            info.AddValue("_ignoreCase", true);
```

Comment:
If I understand you correctly you mean:

1. add the _ignoreCase field OrdinalComparer
2. remove the _ignoreCase field from OrdinalIgnoreCaseComparer
3. inheriting OrdinalIgnoreCaseComparer from OrdinalComparer

Is that right? If yes then I don't think that will solve the issue. When Core deserializes the core/netfx blob it will create an object of type OrdinalComparer (even if the _ignoreCase field is true) and sets the _ignoreCase fields value. The OriginalIgnoreCaseComparer type will never be used. Am I missing something?